src/Stream.php

```diff
@@ -313,4 +310,104 @@ public function getMetadata($key = null)
 
         return $meta[$key] ?? null;
     }
+
+    private static function openZvalStream(string $body)
+    {
+        static $wrapper;
+
+        $wrapper ?? \stream_wrapper_register('Nyholm-Psr7-Zval', $wrapper = \get_class(new class() {
```

Comment:
It's not registered twice, where do you read that? About `??=`, PHP 7.1 rulez. For the "if", I find the current style more readable, because less indentation.

Reply:
I would personally prefer to not do it as any unrelated code MUST never touch wrappers that are not owned by the code.

Reply:
I prefer keeping it, adding some guards against global scope manipulations.